polyphemus/lib/metis/waiver_helper.rb

```diff
@@ -0,0 +1,143 @@
+require 'etna'
+require_relative '../commands'
+
+class MetisWaiverHelper
+  include Polyphemus::WithEtnaClients
+  include Polyphemus::WithLogger
+
+  def initialize(metis_client)
+    @metis_client = metis_client
+  end
+
+  def patient_name_regex(patient_name)
+    /.*\/#{patient_name}-[\w-]+$/
+  end
+
+  def pool_regex(pool_name)
+    /.*\/#{pool_name}$/
+  end
+
+  def restrict_patient_data(patient_name)
+    # Move a folder from release_bucket to restrict_bucket
```

Comment:
These four methods are nearly identical and should probably be refactored into a single method. This whole class could also pretty easily be abstracted to accept some generic model_name and bucket_name arguments and become a move_folder_helper class rather than a waiver_helper class.